scanner/tlstarget.go

```diff
@@ -433,8 +439,26 @@ func (h *CertHostTLSTarget) Dump(hostFh, certFh, chrFh, scsvFh, httpFh *os.File,
 				}
 
 				// Write to certificate-host relation CSV file
-				// [cert_hash, host, port, depth]
-				if ok := chrCsv.Write([]string{sha256Hex, ip, port, h.domain, strconv.Itoa(i)}); ok != nil {
+				// [cert_hash, pub_key_hash, host, port, depth]
+				publicKeyDer, err := x509.MarshalPKIXPublicKey(cert.PublicKey)
+				var sha1PubKey string
+				if err != nil {
+					log.WithFields(log.Fields{
+						"cert_hash":   sha256Hex,
+						"ip":          ip,
+						"server_name": h.domain,
+					}).Error("Could not parse public key")
+					sha1PubKey = ""
+				} else {
+					publicKeyBlock := pem.Block{
+						Type:  "PUBLIC KEY",
+						Bytes: publicKeyDer,
+					}
+
+					sha1PubKey = hex.EncodeToString(getSHA1(pem.EncodeToMemory(&publicKeyBlock)))
```

Comment:
Is `SHA-1(pub_key)` the right thing to store here? If we want to store a hash of the public key, we should use the widely used (e.g. by crt.sh, Censys, Certificate Transparency) hash on the Subject Public Key Info (SPKI). Also, SHA-256 is more suitable than SHA-1.

See [here](https://github.com/mozilla/tls-observatory/blob/aae45faa4006575015d03f6ce1f1732c12d043d2/certificate/certificate.go#L222) for an SHA256-SPKI implementation.

Reply:
originally my intention with sha1 was to reduce storage usage. Going to fix it in order to provide compatibility

Reply:
In addition to using SHA-256 for hashing, we should hash the SPKI instead of only the public key, i.e. use `RawSubjectPublicKeyInfo` instead of `PublicKey` for hashing input. See the link I posted above for a sample implementation.

Reply:
ah I understand. changed it